resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ResourceMethodRegistry.java

```diff
@@ -238,6 +241,25 @@ public void addResourceFactory(ResourceFactory ref, String base, Class<?>[] clas
 
    }
 
+   private Method[] getDeclaredMethods(final Class<?> clazz) {
+      Method[] methodList = new Method[0];
+      try {
+         if (System.getSecurityManager() == null) {
+            methodList = clazz.getDeclaredMethods();
+         } else {
+            methodList = AccessController.doPrivileged(new PrivilegedExceptionAction<Method[]>() {
+               @Override
+               public Method[] run() throws Exception {
+                  return clazz.getDeclaredMethods();
+               }
+            });
+         }
+      } catch (PrivilegedActionException pae) {
+

```

Comment:
Here either the pae exception here should be thrown or documented in the comment why the exception can be ignored.